### DISCUSSION TITLE
missing in the 'defaultValue' prop of either its Controller or useForm

### DISCUSSION THREAD

**Discussion by zhushenhang** [react-hook-form/react-hook-form#3409] [Q&A] [answered]
I set the defaultValue in the Controller, but I received a warning.

img_url is missing in the 'defaultValue' prop of either its Controller (https://react-hook-form.com/api#Controller) or useForm (https://react-hook-form.com/api#useForm)

How can i fix this problem?

my code 
PathForm.js
`
const PathForm = (props) => {
  const classes = useStyles();
  const { handleSubmit, errors, control } = useForm();
  const { onSubmit, pathData } = props;

  const operators = [
    {
      display: <Typography>submit</Typography>,
      size: 'small',
      color: 'primary',
      variant: 'contained',
      type: 'submit'
    }
  ];

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={classes.form}>
      <Controller
        as={StandardInput}
        name="img_url"
        defaultValue={pathData.img_url}
        autoFocus
        fullWidth
        required
        control={control}
        rules={{
          required: true,
          pattern: /^[a-zA-Z]:\\[\u4e00-\u9fa5a-zA-Z\\\w0-9]+$/
        }}
        helperText={errors.img_url && ERROR_MESSAGE}
        error={errors.img_url ? true : false}
        typography="img_url"
      />
    </form>
  );
};
export default PathForm;
`
StandardInput.js 
`
const StandardInput = (props) => {
  const {
    name,
    defaultValue,
    autoFocus,
    helperText,
    margin,
    fullWidth,
    required,
    error,
    type,
    typography,
    typographyClass
  } = props;

  return (
    <React.Fragment>
      <Typography className={typographyClass}>{typography}</Typography>
      <TextField
        name={name}
        defaultValue={defaultValue}
        autoFocus={autoFocus}
        helperText={helperText}
        margin={margin}
        fullWidth={fullWidth}
        required={required}
        error={error}
        type={type}
      />
    </React.Fragment>
  );
};
export default StandardInput;
`

**Comment by bluebill1049** [comment, marked as answer]
- can you see the defaultValue in the component itself?
- provide a codesandbox for us to reproduce the problem.

**Reply by zhushenhang** [reply]
https://codesandbox.io/s/peaceful-tree-419dh?fontsize=14&hidenavigation=1&theme=dark

can you open the codesandbox? 
just simple code

**Reply by bluebill1049** [reply]
`defaultValue={pathData.img_url || null}` ? `undefined` is the issue.

**Reply by zhushenhang** [reply]
thank you for your help.
yes, this method fixes the warning. But I only set a string, why can't StandardInput get props.defaultValue.
like ```defaultValue={"123" || null}```
other components do not have this problem
